### PR TITLE
🎉 New Source: Criteo [Python CDK]

### PR DESCRIPTION
**Pull request by carlonuccio**
## What
This PR adds the new low-code CDK based [Criteo](https://developers.criteo.com/marketing-solutions/reference). Criteo is an advertising company that provides online display advertisements.

## How
This PR uses the lowcode CDK to create a source connector with the following stream:

- [statistic_report](https://developers.criteo.com/marketing-solutions/reference/getadsetreport): Advertisements Report

## Recommended reading order
1. `spec.json`
2. `source.py`
3. `authenticator.py`
4. `streams.py`

## 🚨 User Impact 🚨

-

## Pre-merge Checklist
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:source-criteo:integrationTest`.
- [ ] Code reviews completed
- [ ] Connector version is set to `0.0.1`
    - [ ] `Dockerfile` has version `0.0.1`
- [ ] Documentation updated
    - [x] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

### CODE REVIEW

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
The connector is not using the low-code CDK instead is using the Python CDK.

**Comment by carlonuccio** [comment]
`
$ python -m pytest integration_tests -p integration_tests.acceptance

==================================================================================================== short test summary info ====================================================================================================
SKIPPED [1] ../../bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py:103: The previous connector image could not be retrieved.
SKIPPED [1] ../../bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py:581: The previous connector image could not be retrieved.

Results (188.51s (0:03:08)):
      38 passed
       2 skipped
`

**Comment by carlonuccio** [comment]
```
(.venv) (base) ➜  source-criteo git:(source-criteo) python -m pytest unit_tests
Test session starts (platform: darwin, Python 3.9.12, pytest 6.2.5, pytest-sugar 0.9.7)
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('airbyte-integrations/connectors/source-criteo/.hypothesis/examples')
rootdir: airbyte, configfile: pytest.ini
plugins: hypothesis-6.54.6, requests-mock-1.9.3, mock-3.6.1, sugar-0.9.7, timeout-1.4.2, cov-3.0.0
collected 15 items

 airbyte-integrations/connectors/source-criteo/unit_tests/test_incremental_streams.py::test_cursor_field ✓                                                                                                          7% ▋
 airbyte-integrations/connectors/source-criteo/unit_tests/test_incremental_streams.py::test_supports_incremental ✓                                                                                                 13% █▍
 airbyte-integrations/connectors/source-criteo/unit_tests/test_incremental_streams.py::test_source_defined_cursor ✓                                                                                                20% ██
 airbyte-integrations/connectors/source-criteo/unit_tests/test_incremental_streams.py::test_stream_checkpoint_interval ✓                                                                                           27% ██▋
 airbyte-integrations/connectors/source-criteo/unit_tests/test_source.py::test_check ✓                                                                                                                             33% ███▍
 airbyte-integrations/connectors/source-criteo/unit_tests/test_source.py::test_streams ✓                                                                                                                           40% ████
 airbyte-integrations/connectors/source-criteo/unit_tests/test_streams.py::test_request_params ✓                                                                                                                   47% ████▋
 airbyte-integrations/connectors/source-criteo/unit_tests/test_streams.py::test_parse_response ✓                                                                                                                   53% █████▍
 airbyte-integrations/connectors/source-criteo/unit_tests/test_streams.py::test_request_headers ✓                                                                                                                  60% ██████
 airbyte-integrations/connectors/source-criteo/unit_tests/test_streams.py::test_http_method ✓                                                                                                                      67% ██████▋
 airbyte-integrations/connectors/source-criteo/unit_tests/test_streams.py::test_should_retry[HTTPStatus.OK-False] ✓                                                                                                73% ███████▍
 airbyte-integrations/connectors/source-criteo/unit_tests/test_streams.py::test_should_retry[HTTPStatus.BAD_REQUEST-False] ✓                                                                                       80% ████████
 airbyte-integrations/connectors/source-criteo/unit_tests/test_streams.py::test_should_retry[HTTPStatus.TOO_MANY_REQUESTS-True] ✓                                                                                  87% ████████▋
 airbyte-integrations/connectors/source-criteo/unit_tests/test_streams.py::test_should_retry[HTTPStatus.INTERNAL_SERVER_ERROR-True] ✓                                                                              93% █████████▍
 airbyte-integrations/connectors/source-criteo/unit_tests/test_streams.py::test_backoff_time ✓                                                                                                                    100% ██████████
======================================================================================================= warnings summary ========================================================================================================
airbyte-integrations/connectors/source-criteo/unit_tests/test_incremental_streams.py: 4 warnings
airbyte-integrations/connectors/source-criteo/unit_tests/test_source.py: 1 warning
airbyte-integrations/connectors/source-criteo/unit_tests/test_streams.py: 9 warnings
  /airbyte-integrations/connectors/source-criteo/.venv/lib/python3.9/site-packages/airbyte_cdk/sources/streams/http/http.py:45: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    self._authenticator: HttpAuthenticator = NoAuth()

airbyte-integrations/connectors/source-criteo/unit_tests/test_incremental_streams.py: 4 warnings
airbyte-integrations/connectors/source-criteo/unit_tests/test_source.py: 1 warning
airbyte-integrations/connectors/source-criteo/unit_tests/test_streams.py: 9 warnings
  /airbyte-integrations/connectors/source-criteo/.venv/lib/python3.9/site-packages/deprecated/classic.py:173: DeprecationWarning: Call to deprecated class HttpAuthenticator. (Use requests.auth.AuthBase instead) -- Deprecated since version 0.1.20.
    return old_new1(cls, *args, **kwargs)

-- Docs: https://docs.pytest.org/en/stable/warnings.html

Results (0.39s):
      15 passed
```

File: airbyte-integrations/connectors/source-criteo/acceptance.json [hunk 1]
```diff
@@ -0,0 +1,10248 @@
+[1m============================= test session starts ==============================[0m
```
[the remark this answers is not in the document]
Comment (carlonuccio) [reply airbyte-integrations/connectors/source-criteo/acceptance.json]: removed

**Comment by marcosmarxm** [comment]
Thanks @carlonuccio right now we have a backlog of community pull requests. We're prioritizing alpha and beta bugs over new connectors. I hope to get your contribution to our future sprint. Sorry about the delay.

**Comment by marcosmarxm** [comment]
Hello, our team ended up losing track of reviewing your contribution and we're sorry about that. I'm cleaning up our backlog and closing it now. If you'd like to resubmit it, we'll be very happy and we'll do our best to prioritize the proper review. For new connectors, the project always aims to obtain integration accounts with the implemented service to ensure that the connector works effectively over time. Once again, I apologize, and feel free to send me a message on Slack if you have any further questions.

**Pull request closed without merging**